test/flows/business_coronavirus_support_finder_flow_test.rb

```diff
@@ -0,0 +1,260 @@
+        end
+      end
+    end
+  end
+
+  context "question: self_employed?" do
+    setup do
+      testing_node :self_employed?
+      add_responses business_based?: "england",
+                    business_size?: "0_to_249",
+                    paye_scheme?: "yes"
+    end
+
+    should "render the question" do
+      assert_rendered_question
+    end
+
+    context "next_node" do
+      %w[yes no].each do |response|
+        should "have a next node of non_domestic_property? for a #{response} response" do
+          assert_next_node :non_domestic_property?, for_response: response
+        end
+      end
+    end
+  end
+
+  context "question: non_domestic_property?" do
+    setup do
+      testing_node :non_domestic_property?
+      add_responses business_based?: "england",
+                    business_size?: "0_to_249",
+                    paye_scheme?: "yes",
+                    self_employed?: "yes"
+    end
+
+    should "render the question" do
+      assert_rendered_question
+    end
+
+    context "next_node" do
+      %w[yes no not_sure].each do |response|
+        should "have a next node of sectors? for a #{response} response" do
+          assert_next_node :sectors?, for_response: response
+        end
+      end
+    end
+  end
+
+  context "question: sectors?" do
+    setup do
+      testing_node :sectors?
+      add_responses business_based?: "england",
+                    business_size?: "0_to_249",
+                    paye_scheme?: "yes",
+                    self_employed?: "yes",
+                    non_domestic_property?: "yes"
+    end
+
+    should "render the question" do
+      assert_rendered_question
+    end
+
+    context "next_node" do
+      %w[nurseries retail_hospitality_or_leisure nightclubs_or_adult_entertainment personal_care none].each do |response|
+        should "have a next node of closed_by_restrictions? for a #{response} response" do
+          assert_next_node :closed_by_restrictions?, for_response: [response]
+        end
+      end
+
+      should "have a next node of closed_by_restrictions? for a multiple responses" do
+        assert_next_node :closed_by_restrictions?, for_response: %w[nightclubs_or_adult_entertainment personal_care]
+      end
+    end
+  end
+
+  context "question: sectors?" do
+    setup do
+      testing_node :closed_by_restrictions?
+      add_responses business_based?: "england",
+                    business_size?: "0_to_249",
+                    paye_scheme?: "yes",
+                    self_employed?: "yes",
+                    non_domestic_property?: "yes",
+                    sectors?: %w[none]
+    end
+
+    should "render the question" do
+      assert_rendered_question
+    end
+
+    context "next_node" do
+      %w[local_1 local_2 national none].each do |response|
+        should "have a next node of results for a #{response} response" do
+          assert_next_node :results, for_response: [response]
```

Comment:
```suggestion
          assert_next_node :results, for_response: response
```

Reply:
It's the same here as for `sectors?`. `closed_by_restrictions?` is a checkbox question, so I choose to pass the responses as an array to be consistent with the other test in this context.

Reply:
Would it be better to combine the tests? i.e. rather than have two tests, one for a single response, and another for multiple responses, would it be better, to just have one for all responses? e.g.

```
should "have a next node of results for all responses" do
  assert_next_node :results, for_response: %w[local_1 local_2 national none]
end
```

I could do the same for `sectors?`.

Reply:
@leenagupte yeah I think that would be better! It would assume that if multiple selection works then single selection should too... which I think is fine.